### PR TITLE
ShowOrderDetailsAction handles guest customer orders.

### PR DESCRIPTION
**Pull request by dlobato**
Following conversation #428, this is an initial attempt to allow guest to access order details.

### CODE REVIEW

File: src/Controller/Order/ShowOrderDetailsAction.php [hunk 1]
```diff
@@ -37,20 +37,33 @@ public function __construct(
...
-            ;
-        } catch (TokenNotFoundException $exception) {
-            return $this->viewHandler->handle(View::create(null, Response::HTTP_UNAUTHORIZED));
+        try {
+            if (null !== $user) {
+                $order = $this
+                    ->placedOrderQuery
+                    ->getOneCompletedByCustomerEmailAndToken($user->getEmail(), (string) $request->attributes->get('tokenValue'))
+                ;
+            } else {
```
Comment (lchrusciel) [inline src/Controller/Order/ShowOrderDetailsAction.php]: Can we extract this logic to a private function and use early return?

Comment (dlobato) [reply src/Controller/Order/ShowOrderDetailsAction.php]: Sure

File: src/Controller/Order/ShowOrderDetailsAction.php [hunk 2]
```diff
@@ -37,20 +38,39 @@ public function __construct(
...
-                ->getOneCompletedByCustomerEmailAndToken($user->getCustomer()->getEmail(), (string) $request->attributes->get('tokenValue'))
-            ;
-        } catch (TokenNotFoundException $exception) {
-            return $this->viewHandler->handle(View::create(null, Response::HTTP_UNAUTHORIZED));
+        try {
+            $order = $this->getPlacedOrderView(
+                (string) $request->attributes->get('tokenValue'),
+                $user
+            );
         } catch (\InvalidArgumentException $exception) {
```
Comment (mamazu) [inline src/Controller/Order/ShowOrderDetailsAction.php]: I am not really sure, do we want to have a better handling here if the token belongs to another user. So a 404 for an unknown token and a 403 Unauthorized for tokens that need login maybe.

File: src/Controller/Order/ShowOrderDetailsAction.php [hunk 3]
```diff
@@ -37,20 +38,39 @@ public function __construct(
 
     public function __invoke(Request $request): Response
     {
-        try {
+        $groups = [GroupsExclusionStrategy::DEFAULT_GROUP];
```
Comment (mamazu) [inline src/Controller/Order/ShowOrderDetailsAction.php]: Very nice idea to handle this logic with view groups. Didn't know that was an option. :+1:

**Comment by mamazu** [comment]
Then this should be ready to be merged.

**Comment by dlobato** [comment]
Yes

**Comment by mamazu** [comment]
Thank you, David! :tada:

**Pull request merged**